### PR TITLE
Copy properties when cloning a ModelObject

### LINKED ISSUE #107: ModelObject#clone does not copy properties
Currently, [ModelObject#clone](https://github.com/spdx/Spdx-Java-Library/blob/master/src/main/java/org/spdx/library/model/ModelObject.java#L993-L999) creates a clone of the instance it is called on inside the passed model store. It does not copy any properties of the original object. The properties of the new object will be:
- Empty in case the object did not exist already in the new store
- The corresponding properties from the new store if the object already existed in there

In the discussion in https://github.com/spdx/Spdx-Java-Library/pull/106, it turned out that this may be unintended and properties should actually be copied over.
In that case, we should probably fail in case the object already exists in the new store, to avoid accidentally overwriting data (or pass a `overwrite` flag).

Note: The only usage of `clone` in the library seems to be inside a test, so changing it won't cause problems in the library. It could be considered a breaking change to public API, however...

### PR DESCRIPTION
**Pull request by goneall**
Fixes #107 

In addition to copying the properties in clone, I also added some additional checks.

Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

### CODE REVIEW

File: src/main/java/org/spdx/library/model/ModelObject.java [hunk 1]
```diff
@@ -1002,8 +1002,20 @@ public boolean equals(Object o) {
 	 * @return
 	 */
 	public ModelObject clone(IModelStore modelStore) {
+		if (Objects.isNull(this.copyManager)) {
+			throw new RuntimeException("A copy manager must be provided to clone");
```
Comment (nicoweidner) [inline src/main/java/org/spdx/library/model/ModelObject.java:1006]: nit: Maybe `IllegalStateException` would be a bit more precise?

**Review by nicoweidner — APPROVED** [review]
Looks good to me, thanks for putting this in so quickly!

I just left a few nitpicky suggestions to use more specific exception classes rather than generic `RuntimeException`s

File: src/main/java/org/spdx/library/model/ModelObject.java (hunk 1 above)
Comment (goneall) [reply src/main/java/org/spdx/library/model/ModelObject.java]: Agree - changed to `IllegalStateException`

File: src/main/java/org/spdx/library/model/ModelObject.java [hunk 2]
```diff
@@ -1002,8 +1002,20 @@ public boolean equals(Object o) {
...
+		}
+		if (this.modelStore.equals(modelStore)) {
+			throw new RuntimeException("Can not clone to the same model store");
```
[the remark this answers is not in the document]
Comment (goneall) [reply src/main/java/org/spdx/library/model/ModelObject.java]: Agree - changed to `IllegalStateException`

File: src/main/java/org/spdx/library/model/ModelObject.java [hunk 3]
```diff
@@ -1002,8 +1002,20 @@ public boolean equals(Object o) {
...
+		}
+		Objects.requireNonNull(modelStore, "Model store for clone must not be null");
+		if (modelStore.exists(this.documentUri, this.id)) {
+			throw new RuntimeException("Can not clone - "+this.id+" already exists.");
```
[the remark this answers is not in the document]
Comment (goneall) [reply src/main/java/org/spdx/library/model/ModelObject.java]: Agree - changed to `IllegalStateException`

**Pull request merged**